README.md

```diff
@@ -53,3 +53,133 @@ data:
+
+## Development
+
+### Set up local OpenShift cluster
+
+Methods include:
+- [MiniShift](https://github.com/minishift/minishift)
+- [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/overview)
+- [Integration OpenShift Cluster Manager](https://qaprodauth.cloud.redhat.com/openshift/?env=integration)
+
+### Deploy dependencies
+
+[Hive](https://github.com/openshift/hive/) CRDs need to be installed on the cluster.
+
+Clone [hive repo](https://github.com/openshift/hive/) and run
+
+```terminal
+$ git clone https://github.com/openshift/hive.git
+$ oc apply -f hive/config/crds
+```
+
+Install the `DeadMansSnitchIntegration` CRD, create the operator namespace and other operator dependencies:
+
+```terminal
+$ oc apply -f deploy/crds/deadmanssnitch.managed.openshift.io_deadmanssnitchintegrations_crd.yaml
+$ oc new-project deadmanssnitch-operator
+$ oc apply -f deploy/role.yaml
+$ oc apply -f deploy/service_account.yaml
+$ oc apply -f deploy/role_binding.yaml
+$ oc apply -f deploy/prometheus-k8s-role.yaml
+$ oc apply -f deploy/prometheus-k8s-rules.yaml
+$ oc apply -f deploy/prometheus-k8s-rolebinding.yaml
+```
+
+Create a secret which will contain the DeadMansSnitch API Key and Hive Cluster Tag. 
+
+You will require an API Key signed up to a DeadMansSnitch plan that allows for enhanced snitch intervals (the "Private Eye" plan). You can alternatively test the `deadmanssnitch-oeprator` by signing up to the free tier DeadMansSnitch plan (limited to 1 snitch), but doing so will require you to customize the snitch interval from `15_minute` to `hourly`. This can be performed in [deadmanssnitchintegration_controller.go](pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go) 
+
+Adjust the example below and apply the file with `oc apply -f <file>`. Note that the values for `hive-cluster-tag` and `deadmanssnitch-api-key` need to be base64 encoded. This can be performed using `echo -n <text> | base64`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: deadmanssnitch-api-key
+  namespace: deadmanssnitch-operator
+data:
+  hive-cluster-tag: <value>
+  deadmanssnitch-api-key: <value>
+```
+
+### Define a DeadMansSnitchIntegration
+
+Create a `DeadMansSnitchIntegration` CR which will be used to identify clusters to apply DMS to.
+
+The example below will target `clusterdeployment`s that have a `api.openshift.com/test` label set to `"true"`. Apply it using `oc apply -f <file>`.
+
+```yaml
+apiVersion: deadmanssnitch.managed.openshift.io/v1alpha1
+kind: DeadmansSnitchIntegration
+metadata:
+  finalizers:
+  - dms.managed.openshift.io/deadmanssnitch-osd
+  name: test-dmsi
+  namespace: deadmanssnitch-operator
+spec:
+  clusterDeploymentSelector:
+    matchExpressions:
+    - key: api.openshift.com/test
+      operator: In
+      values:
+      - "true"
+  dmsAPIKeySecretRef:
+    name: deadmanssnitch-api-key
+    namespace: deadmanssnitch-operator
+  snitchNamePostFix: ""
```

Comment:
```suggestion
  snitchNamePostFix: "test"
```

It's only optional because OSD today doesn't use a postfix.  In examples we should include a value.